ArkKit/ark-ui-kit-renderer/view-components/input/UIKitJoystick.swift

```diff
@@ -88,17 +83,16 @@ final class UIKitJoystick: UIView, UIKitRenderable, PanRenderable {
 
 extension UIKitJoystick {
     func applyModifiers(modifierInfo: JoystickCanvasComponent) -> Self {
-        let defaultPanHandler = { (_: Double, _: Double) in }
-
+        if let onPanStartHandler = modifierInfo.onPanStartDelegate {
```

Comment:
i think @chownces implemented the `modify` and `if` methods so that the component structs are immutable.

might be better to do to preserve immutability. 
```swift
var newJoystick = self
// replace self with newJoystick
```